src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -723,21 +731,22 @@ private String getQualifiedTypeDeclarationName(DetailAST typeDeclAst) {
      * @param className className
      * @return the qualified name of type declaration by combining {@code packageName},
      *         {@code outerClassQualifiedName} and {@code className}
+     * @noinspection CallToStringConcatCanBeReplacedByOperator
```

Comment:
Why is this change ok? It seems odd.

Reply:
```
if (packageName == null) {
}
```
pit mutated it to false, so 
```
qualifiedClassName = packageName + PACKAGE_SEPARATOR + className;
```
resulted in `null.SomeClass`, so to get an exception I replaced it with `concat` to kill the mutation.
Actually, this method was picked from `FinalClassCheck`, it works for this check also but to kill some mutations some changes need to be made, there is difference between the implementation of both these check hence this was required.

Reply:
Why can't we create a case to kill it without resulting to this which CI is set to avoid doing? We can't create a special package or such possibly matching `null.SomeClass`?

Reply:
There were quite a few mutations killed by using `null`, I had seen cases which required using null instead of `""`, but I will look into it again and try to do it with `""`, will post the results here.

Reply:
I am not against removing `null` so if that makes other mutations easier then ok. Why can't we use `null` someway to kill it for `packageName == null`. 
The result of the if/else is a big difference between `SomeClass` and `null.SomeClass`.

Reply:
Done